lib/dataset/dataset.cpp

```diff
@@ -185,12 +186,21 @@ void Dataset::setData(const std::string &name, DataArray data) {
   for (auto &&[dim, coord] : data.coords())
     if (m_coords.find(dim) == m_coords.end() &&
         std::find(to_skip.begin(), to_skip.end(), dim) == to_skip.end())
-      setCoord(dim, std::move(coord));
+      setCoord(dim, coord);
   // Attrs might be shadowed by a coord, but this cannot be prevented in
   // general, so instead of failing here we proceed (and may fail later if
   // meta() is called).
-  item.attrs() = std::move(data.attrs());
-  item.masks() = std::move(data.masks());
+  item.attrs() = data.attrs();
+  item.masks() = data.masks();
+}
+
+/// Like setData but allow inserting into a default-initialized dataset.
+void Dataset::setDataInit(const std::string &name, const DataArray &data) {
```

Comment:
Something about this mechanism feels wrong, is it because it is a hack to get back size-changing `setData`? When suggesting it in Slack I had in mind that invalid `Dataset` would only be used for overwriting (using `operator=`), not for this. Should probably have a chat to clarify our understanding.

Reply:
Maybe you can document this more clearly?